packit/utils/changelog_helper.py

```diff
@@ -0,0 +1,124 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import shutil
+import logging
+from typing import Optional
+
+from packit.actions import ActionName
+from packit.config.common_package_config import CommonPackageConfig
+from packit.distgit import DistGit
+import packit.upstream
+
+logger = logging.getLogger(__name__)
+
+
+class ChangelogHelper:
+    def __init__(
+        self,
+        upstream: "packit.upstream.Upstream",
+        downstream: Optional[DistGit] = None,
+        package_config: Optional[CommonPackageConfig] = None,
+    ) -> None:
+        self.up = upstream
+        self.dg = downstream
+        self.package_config = package_config
+
+    @property
+    def entry_from_action(self) -> Optional[str]:
+        """
+        Runs changelog-entry action if present and returns string that can be
+        used as a changelog entry.
+
+        Returns:
+            Changelog entry or `None` if action is not present.
+        """
+        messages = self.up.get_output_from_action(ActionName.changelog_entry)
+        if not messages:
+            return None
+
+        return "\n".join(map(lambda line: line.rstrip(), messages))
```

Comment:
nicely done wrt `.rstrip()`, since the entries may contain important leading whitespace